plugins/misc/w_helicity.h

```diff
@@ -0,0 +1,186 @@
+  p4lab_aLep.SetPtEtaPhiM(arg[12], arg[13], arg[14], arg[15]);
+  p4lab_pW.SetPtEtaPhiM(arg[16], arg[17], arg[18], arg[19]);
+  p4lab_aW.SetPtEtaPhiM(arg[20], arg[21], arg[22], arg[23]);
+  p4lab_pB.SetPtEtaPhiM(arg[24], arg[25], arg[26], arg[27]);
+  p4lab_aB.SetPtEtaPhiM(arg[28], arg[29], arg[30], arg[31]);
+
+  // the necessary boosting such that lepton ~ top spin vector
+  const TLorentzVector p4lab_TT( p4lab_pTop + p4lab_aTop ); //calculates the t-tbar system 4-momenta
+  // const TLorentzVector p4lab_eb( p4lab_aLep + p4lab_pB ); //calculates the t-tbar system 4-momenta
+
+
+  const auto f_zmf_tt = [&p4lab_TT] (TLorentzVector p4) {
+    // function that boost a 4-vector p4 to the zmf-tt sytem
+    // p4lab_TT.BoostVector() returns the beta vector (relative speed) of p4lab_TT (TTbar system) with respect to the lab system
+    // p4.Boost(-1*vec{beta}) peforms the direct lorentz transform from the lab system to the TTbar system, where beta is the relative speed between them
+    p4.Boost( -1. * p4lab_TT.BoostVector() ); //Boost() is the inverse Lorentz transformation, therefore the -1 factor to the relative velocity (beta vector) to give the direct Lorentz transform.
+    return p4;
+  };
+  // const TLorentzVector p4hel_pTop = f_zmf_tt(p4lab_pTop);
+  // const TLorentzVector p4hel_aTop = f_zmf_tt(p4lab_aTop);
+
+  // const auto f_zmf_top = [&f_zmf_tt] (const TLorentzVector &wboson, const TLorentzVector &top) {
+  //   // function that boost a 4-vector W to the direction of the 4-momentum of the TT sytem, and then reboost it to the direction of the 4-vector top
+  //   auto p4 = f_zmf_tt(wboson); // boost a 4-vector wboson to the direction of the TT zmf sytem
+  //   p4.Boost( -1. * top.BoostVector() ); // inverse of boosting the 4-vec top to the system where p4 is defined, i.e the tt-zmf. I.e. boost p4 from tt_zmf to the top zmf
+  //   return p4;
+  // };
+
+
+  const auto f_zmf_w = [&f_zmf_tt] (const TLorentzVector &daugther, const TLorentzVector &wboson) {
+    // function that boost a 4-vector u4 to the direction of the 4-momentum of the TT sytem, and then reboost it to the direction of the 4-vector w
+    auto p4 = f_zmf_tt(daugther);
+    p4.Boost( -1. * wboson.BoostVector() );
+    return p4;
+  };
+
+  // const TLorentzVector p4top_pW = f_zmf_top(p4lab_pW, p4hel_pTop); //directon of flight of w in the t-ZMF
+  // const TLorentzVector p4top_aW = f_zmf_top(p4lab_aW, p4hel_aTop);
+
+  const TLorentzVector p4hel_pW = f_zmf_tt(p4lab_pW); //directon of flight of W+ in the tt-ZMF
+  const TLorentzVector p4hel_aW = f_zmf_tt(p4lab_aW);
+
+  const TLorentzVector p4hel_aLep = f_zmf_w(p4lab_aLep, p4hel_pW); //directon of flight of anti-lepton in the w-ZMF
+  const TLorentzVector p4hel_pLep = f_zmf_w(p4lab_pLep, p4hel_aW);
+
+  const TLorentzVector p4hel_pB = f_zmf_w(p4lab_pB, p4hel_pW); //directon of flight of b-quark in the w-ZMF
+  const TLorentzVector p4hel_aB = f_zmf_w(p4lab_aB, p4hel_aW);
+
```

Comment:
does this mean that w is evaluated in ttbar zmf but l/b are in lab frame, before directly being brought to w zmf (from ttbar zmf)? probably should mention the reference for doing it this way in the code

Reply:
Yes, precisely. I will change that, such that both arguments of the f_zmf_w are in the lab frame

Reply:
also, without the ttbar zmf tops are no longer used, and so can be removed from function arg.

Reply:
Indeed you are right that it is not necessary anymore. They are all remaining from the old code, where I was also using the the other definitions of cos(theta*) that involve top zmf. In that case, ttbar zmf function was also not necessary, but convenient codewise.
Now all of that is updated and ttbar zmf is not there anymore